src/main/java/nextstep/subway/domain/Sections.java

```diff
@@ -0,0 +1,141 @@
+package nextstep.subway.domain;
+
+import nextstep.subway.domain.exception.SectionExceptionMessages;
+import org.springframework.dao.DataIntegrityViolationException;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Embeddable;
+import javax.persistence.OneToMany;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+@Embeddable
+public class Sections {
+
+    @OneToMany(mappedBy = "line", cascade = {CascadeType.PERSIST, CascadeType.MERGE}, orphanRemoval = true)
+    private List<Section> sections = new ArrayList<>();
+
+    public Sections() {
+    }
+
+    public Sections(Section section) {
+        this.sections.add(section);
+    }
+
+    public void addSection(Line line, Section section) {
+        validateAddSection(line, section);
+
+        if (!hasSections()) {
+            sections.add(section);
+            line.changeFirstStation(section.getUpStation());
+            line.changeLastStation(section.getDownStation());
+            return;
+        }
+
+        if (section.isNewFirstSection()) {
+            sections.add(section);
+            line.changeFirstStation(section.getUpStation());
+            return;
+        }
+
+        if (section.isNewLastSection()) {
+            sections.add(section);
+            line.changeLastStation(section.getDownStation());
+            return;
+        }
+
+        addBetweenSection(line, section);
+    }
+
+    public boolean hasSections() {
+        return sections.size() > 0;
```

Comment:
내부적으로는 sections.isEmpty()라는 메서드가 있어서 이를 이용해보는건 어떨까 싶어요 
```suggestion
        return !sections.isEmpty()
```